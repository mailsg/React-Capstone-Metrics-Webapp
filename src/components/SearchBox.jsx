import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { searchQuery } from '../redux/slice/meals/mealsSlice';

const SearchBox = () => {
  const dispatch = useDispatch();
  const [searchTerm, setSearchTerm] = useState('');

  const handleSearch = (event) => {
    const { value } = event.target;
    setSearchTerm(value);
    dispatch(searchQuery(value));
  };

  return (
    <div className="search-box">
      <input
        type="text"
        placeholder="Search meals.."
        value={searchTerm}
        onChange={handleSearch}
      />
    </div>

  );
};

export default SearchBox;
