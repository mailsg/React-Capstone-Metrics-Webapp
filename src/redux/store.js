import { configureStore } from '@reduxjs/toolkit';
import mealsReducer from './slice/meals/mealsSlice';

const store = configureStore({
  reducer: {
    meals: mealsReducer,
  },
});

export default store;
