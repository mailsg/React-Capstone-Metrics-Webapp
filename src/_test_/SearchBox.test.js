import '@testing-library/jest-dom';
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/store';
import SearchBox from '../components/SearchBox';

describe('SearchBox', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <SearchBox />
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
