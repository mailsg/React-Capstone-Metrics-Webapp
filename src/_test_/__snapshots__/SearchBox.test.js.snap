// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox renders correctly 1`] = `
<div
  className="search-box"
>
  <input
    onChange={[Function]}
    placeholder="Search meals.."
    type="text"
    value=""
  />
</div>
`;
